lib/client.js

```diff
@@ -1901,6 +1902,71 @@ MatrixClient.prototype.scrollback = function(room, limit, callback) {
     return defer.promise;
 };
 
+/**
+ * Take an EventContext, and back/forward-fill results.
+ *
+ * @param {module:models/event-context.EventContext} eventContext  context
+ *    object to be updated
+ * @param {boolean} backwards  true to fill backwards, false to go forwards
+ *
+ * @return {module:client.Promise} Resolves: updated EventContext object
+ * @return {module:http-api.MatrixError} Rejects: with an error response.
+ */
+MatrixClient.prototype.paginateEventContext = function(eventContext, backwards) {
+    // TODO: we should implement a backoff (as per scrollback()) to deal more
+    // nicely with HTTP errors.
+
+    var token = eventContext.getPaginateToken(backwards);
+    if (!token) {
+        // no more results.
+        throw new Error("No paginate token");
+    }
+
+    var dir = backwards ? 'b' : 'f';
+    var pendingRequest = eventContext._paginateRequests[dir];
+
+    if (pendingRequest) {
+        // already a request in progress - return the existing promise
+        return pendingRequest;
+    }
+
+    var path = utils.encodeUri(
+        "/rooms/$roomId/messages", {$roomId: eventContext.getEvent().getRoomId()}
+    );
+    var params = {
+        from: token,
+        limit: 30,
```

Comment:
It would be worth moving this to be configurable. Might be worth having the function signature be `paginateEventContext(eventContent, opts)` which has `opts.backwards` and `opts.limit`?